### ISSUE TITLE
Polygon editing tool improvements

### ISSUE THREAD

**Issue by bjorn** [bjorn/tiled#1487]
I implemented the polygon editing tool years ago and since then it hasn't received much love. However, there are several improvements that could be implemented to make it more useful:

* Allow rotating of selected nodes
* Allow scaling of selected nodes
* Allow converting between polyline and polygon (open vs. closed)
* Allow extending a polyline or polygon with more nodes
* Add a faster way to split segments, by dragging from them or double-clicking them (splitting at interacted location).

For other possible improvements, see also #117, #145, #404, #762 and #882.

**Comment by ketanhwr** [comment]
@bjorn can we discuss more about this tool here? I was going through all the issues that you've mentioned, and it seemed to me that a lot of changes are to be made. That's why I was considering revamping this tool.

**Comment by bjorn** [comment]
@ketanhwr Sure. So far it seemed to me that these issues would all mean to extend the tool, but some things may raise questions. Please go into more detail about the things you consider needing a lot of changes.

**Comment by ketanhwr** [comment]
> - Allow rotating of selected nodes
> - Allow scaling of selected nodes

What does these two mean exactly?

**Comment by Ablu** [comment]
> What does these two mean exactly?

I can think of those to, for example, rotate or scale a subset of the vertices of a polygon. I guess this is mostly useful for larger / complex polygons.

**Comment by bjorn** [comment]
Just see for example how when you select multiple objects, you can rotate and scale them as a group. The same functionality should be available for polygon nodes.

**Comment by JoeCreates** [comment]
I think an important feature which is technically here but kind of gets lost among other issues in #762 is to allow the editing of individual points in properties. It would be useful to map the arrow keys to moving the currently selected node(s) in pixels, too.

A new suggestion is to allow for snapping to angles, e.g. 90, 45 degrees, or gradients of 1/2/3, perhaps with shift key.

I'm in a situation right now where I have no good option to make a straight flat line because my object is not aligned to the grid. I cannot preposition one of my nodes to keep a line flat and put it in the right place for a moving platform to move to the correct place.

![image](https://user-images.githubusercontent.com/2540732/61733013-d3a40a00-ad76-11e9-9a78-b93ec89ed163.png)

**Comment by bjorn** [comment]
> I'm in a situation right now where I have no good option to make a straight flat line because my object is not aligned to the grid. I cannot reposition one of my nodes to the left while maintaining a flat line and put it in the right place for a moving platform (for which this line defines a path) to move to the correct place.

Right, in this case at the moment it could help to zoom in and to enable snapping to pixels. But I definitely agree that it would be good to add angle snapping and manual editing of point coordinates!

**Comment by anaanook** [comment]
> I think an important feature which is technically here but kind of gets lost among other issues in #762 is to allow the editing of individual points in properties. It would be useful to map the arrow keys to moving the currently selected node(s) in pixels, too.

adding a +1 here as I've been hitting this.  There's currently inconsistency between what the arrow keys do when in select objects mode vs edit polygons mode - using the arrow keys to adjust positions.  Arrow key adjustments in the polygon editing tool would be a fantastic addition.

**Comment by jjppof** [comment]
> I think an important feature which is technically here but kind of gets lost among other issues in #762 is to allow the editing of individual points in properties. It would be useful to map the arrow keys to moving the currently selected node(s) in pixels, too.

That would be a very awesome feature, mainly if we could use shift!
